### PR TITLE
Sample api

### PR DESCRIPTION
**Pull request by praminda**

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag [hunk 1]
```diff
@@ -426,4 +426,39 @@ var isScopeExist = function (scope, username) {
...
+var getGatewayURL = function (api) {
+    var carbon = require('carbon');
+    var gatewayURL, envs, result, log = new Log(),
+            provider = jagg.module("manager").getAPIProviderObj();
+    try {
+        envs = provider.getEnvironments();
+        if (log.isDebugEnabled()) {
+            log.debug("Invoke getEnvironments()" );
+        }
+        for(var i = 0; i < envs.length; i++) {
```
Comment (lalaji) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: Adda null check for envs variable,before the for loop.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag [hunk 2]
```diff
@@ -426,4 +426,39 @@ var isScopeExist = function (scope, username) {
...
+            if(envs[i].apiConsole == true) {
+                result = envs[i].serverURL;
+            }
+        }
+        // Check if carbon.server.address url and gateway url is the same and use gateway url as 'localhost'
+        // to avoid ssl host name verification error on localhost
+        var resultComponents = result.split(/:\/{2}|:|\//);
```
Comment (lalaji) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: Better to add null check for result variable as well,before do split operation.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag [hunk 3]
```diff
@@ -426,4 +426,39 @@ var isScopeExist = function (scope, username) {
...
+        var carbonServerAddr = carbon.server.address(resultComponents[0]);
+        var serverComponents = carbonServerAddr.split(/:\/{2}|:|\//);
+        if(serverComponents[1] == resultComponents[1]) {
```
Comment (lalaji) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: Why this logic needed..?Can't we just set the ip without localhost?

File: components/apimgt/samples/org.wso2.carbon.apimgt.samples.calculator/src/main/webapp/index.jsp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<html>
+<body>
+<h2>WSO2 API Manager - Sample calculator api</h2>
```
Comment (chamindias) [inline components/apimgt/samples/org.wso2.carbon.apimgt.samples.calculator/src/main/webapp/index.jsp]: Guess this should be "Sample calculator API"? - slight capitalisation issue.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag (hunk 3 above)
Comment (praminda) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: There was a ssl host name verification error when it is not localhost

File: components/apimgt/samples/org.wso2.carbon.apimgt.samples.calculator/src/main/webapp/index.jsp (hunk 4 above)
Comment (praminda) [inline components/apimgt/samples/org.wso2.carbon.apimgt.samples.calculator/src/main/webapp/index.jsp]: Fixed in #1672

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag (hunk 2 above)
Comment (praminda) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: Fixed in #1672

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag (hunk 1 above)
Comment (praminda) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/modules/api/add.jag]: Fixed in #1672

**Pull request merged**